execution/gethexec/express_lane_service.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2024-2025, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package gethexec
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/arbitrum_types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/lru"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/offchainlabs/nitro/solgen/go/express_lane_auctiongen"
+	"github.com/offchainlabs/nitro/timeboost"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/pkg/errors"
+)
+
+type expressLaneControl struct {
+	sequence   uint64
+	controller common.Address
+}
+
+type expressLaneService struct {
+	stopwaiter.StopWaiter
+	sync.RWMutex
+	auctionContractAddr      common.Address
+	initialTimestamp         time.Time
+	roundDuration            time.Duration
+	auctionClosing           time.Duration
+	chainConfig              *params.ChainConfig
+	logs                     chan []*types.Log
+	seqClient                *ethclient.Client
+	auctionContract          *express_lane_auctiongen.ExpressLaneAuction
+	roundControl             lru.BasicLRU[uint64, *expressLaneControl]
+	messagesBySequenceNumber map[uint64]*timeboost.ExpressLaneSubmission
+}
+
+func newExpressLaneService(
+	auctionContractAddr common.Address,
+	sequencerClient *ethclient.Client,
+	bc *core.BlockChain,
+) (*expressLaneService, error) {
+	chainConfig := bc.Config()
+	auctionContract, err := express_lane_auctiongen.NewExpressLaneAuction(auctionContractAddr, sequencerClient)
+	if err != nil {
+		return nil, err
+	}
+	roundTimingInfo, err := auctionContract.RoundTimingInfo(&bind.CallOpts{})
+	if err != nil {
+		return nil, err
+	}
+	initialTimestamp := time.Unix(int64(roundTimingInfo.OffsetTimestamp), 0)
+	roundDuration := time.Duration(roundTimingInfo.RoundDurationSeconds) * time.Second
+	auctionClosingDuration := time.Duration(roundTimingInfo.AuctionClosingSeconds) * time.Second
```

Comment:
`roundTimingInfo` is retrieved from the smart contract, which is the source of truth we should adhere to. Unless we want to add another configuration to check against this single source of truth, I generally prefer less configuration, as it's less error-prone, but I'm open to changing if that's what we prefer!

Reply:
Agreed that we should rely on the contract here as the source of truth

Reply:
My point is that the code is broken if it's anything other than 1 minute. The contract is the source of truth so we should check that the settings on the contract matches the assumptions we're currently making in the code, and assert if they are violated because otherwise the code will behave in unexpected ways.